Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,320 @@
+        refine mem_properDivisors.mpr ?_
+        constructor
+        exact Dvd.dvd.mul_right han m
+        refine lt_mul_of_lt_of_one_le ?_ hm
+        simp_all only [gt_iff_lt, le_mul_iff_one_le_right, mem_properDivisors, true_and, ne_eq]
+      · have ham : a ∣ m := by
+          simp_all only [gt_iff_lt, le_mul_iff_one_le_right, mem_properDivisors]
+        refine mem_properDivisors.mpr ?_
+        constructor
+        exact Dvd.dvd.mul_left ham n
+        refine lt_mul_of_one_le_of_lt hn ?_
+        simp_all only [gt_iff_lt, le_mul_iff_one_le_right, le_mul_iff_one_le_left,
+          mem_properDivisors, true_and, ne_eq]
+
+lemma divisors_eq_proper_union_self (hn : 0 < n) :
+    divisors n = properDivisors n ∪ {n} := by
+      dsimp [divisors, properDivisors]
+      ext a
+      simp [*]
+      constructor
+      <;> rintro ⟨⟨h1, h2⟩, h3⟩
+      · omega
+      · omega
+      · simp only [*, lt_add_iff_pos_right, zero_lt_one, and_true, dvd_refl]
+        omega
+
+lemma deficient_not_abundant_or_perfect (hn : 0 < n) :
+    Deficient n ↔ ¬ Abundant n ∧ ¬ Perfect n := by
+      dsimp [Perfect, Abundant, Deficient]
+      omega
+
+lemma perfect_not_abundant_or_deficient (hn : 0 < n) :
+    Perfect n ↔ ¬ Abundant n ∧ ¬ Deficient n := by
+      dsimp [Perfect, Abundant, Deficient]
+      omega
+
+lemma abundant_not_perfect_or_deficient (hn : 0 < n) :
+    Abundant n ↔ ¬ Perfect n ∧ ¬ Deficient n := by
+      dsimp [Perfect, Abundant, Deficient]
+      omega
+
+/- All numbers are either deficient, perfect, or abundant -/
+theorem deficient_or_perfect_or_abundant (hn : 0 < n) :
+    Deficient n ∨ Abundant n ∨ Perfect n := by
+      dsimp [Perfect, Abundant, Deficient]
+      omega
+
+theorem perfect_pseudoperfect (h : Perfect n) : Pseudoperfect n := by
+  rcases h with ⟨h1, h2⟩
+  constructor
+  · exact h2
+  · use properDivisors n
+    constructor
+    · exact mem_powerset_self (properDivisors n)
+    · exact h1
+
+theorem prime_not_abundant (h : Prime n) : ¬ Abundant n := by
+  intro h1
+  have h2 : ∑ i in properDivisors n, i = 1 := by exact sum_properDivisors_eq_one_iff_prime.mpr h
+  have h3 : n > 1 := by exact Prime.one_lt h
+  have h4 : ∑ i in properDivisors n, i > 1 := by exact Nat.lt_trans h3 h1
+  omega
+
+theorem prime_not_weird (h : Prime n) : ¬ WeirdNumber n := by
+  simp only [Nat.WeirdNumber, not_and_or]
+  left
+  exact prime_not_abundant n h
+
+theorem not_pseudoperfect_not_perfect (h : ¬ Pseudoperfect n) : ¬ Perfect n := by
+  revert h
+  rw [not_imp_not]
+  exact perfect_pseudoperfect n
+
+theorem prime_not_pseudoperfect (h : Nat.Prime n) : ¬ Nat.Pseudoperfect n := by
+  rw [not_pseudoperfect_iff_forall]
+  have h1 : powerset (properDivisors n) = {∅, {1}} := by
+    rw [Prime.properDivisors h]
+    exact rfl
+  have h2 : n > 1 := by exact Prime.one_lt h
+  rw [h1]
+  right
+  rintro s hs
+  have hs' : s = ∅ ∨ s = {1} := by exact List.mem_pair.mp hs
+  rcases hs' with hs1 | hs2
```

Comment:
Probably, you can inline `hs'`, by writing `  rcases (List.mem_pair.mp hs) with hs1 | hs2`